File: Queue and  Stacks/queueLinkedList.py
class _Node:
    __slots__ = '_element','_next' 
    def __init__(self,element,next):
        self._element=element 
        self._next =next 
class QueuesLinked():
    def __init__(self):
        self._head=None 
        self._tail=None
        self._size=0 
    def __len__(self):
        return self._size 
    def isempty(self):
        return self._size==0 
    def enqueue(self,e):
        newest=_Node(e,None)
        if self.isempty():
            self._head=newest
        else:
            self._tail._next=newest 
        self._tail=newest 
        self._size+=1
    def dequeue(self):
        if self.isempty():
            print("Queue is empty") 
            return 
        e=self._head._element 
        self._head=self._head._next 
        self._size -=1 
        if self.isempty():
            self._tail=None 
        return e 
    def first(self):
        if self.isempty():
            print("Queue is empty") 
            return 
        return self._head._element 
    def display(self):
        p=self._head 
        while p:
            print(p._element ,end="<--")
            p=p._next 
        print()




Q = QueuesLinked()
Q.enqueue(5)
Q.enqueue(3)
print('Queue:')
Q.display()
print('Queue Length:', len(Q))
ele = Q.dequeue()
print('Queue:')
Q.display()
print('Queue Length:', len(Q))
print('Removed Element:',ele)
print('Is Queue Empty:',Q.isempty())
Q.enqueue(7)
print('Queue:')
Q.display()
print('Queue Length:', len(Q))
Q.enqueue(12)
print('Queue:')
Q.display()
print('Queue Length:', len(Q))
ele = Q.dequeue()
print('Queue:')
Q.display()
print('Queue Length:', len(Q))
print('Removed Element:',ele)
print('First Element:',Q.first())

File: Queue and  Stacks/QueueLinkedList.go
package main

import "fmt"

type Node struct {
	val  int
	next *Node
}
type LinkedList struct {
	head *Node
	tail *Node
	size int
}

func (L *LinkedList) isempty() bool {
	return L.size == 0
}
func (L *LinkedList) display() {
	p := L.head
	for p != nil {
		fmt.Printf("%d <-", p.val)
		p = p.next
	}
	fmt.Println("")
}
func (L *LinkedList) enque(e int) {
	newest := &Node{e, nil}
	if L.isempty() {
		L.head = newest
	} else {
		L.tail.next = newest

	}
	L.tail = newest
	L.size++

}
func (L *LinkedList) deque() int {
	if L.isempty() {
		fmt.Println("Queue is empty")
		return 0
	}
	e := L.head.val
	L.head = L.head.next
	L.size--
	if L.isempty() {
		L.tail = nil
	}
	return e

}
func (L *LinkedList) top() int {
	if L.isempty() {
		fmt.Println("Queue is empty")
		return 0
	}
	e := L.tail.val
	return e
}

func main() {
	n := LinkedList{}
	fmt.Println(n)
	fmt.Println(n.isempty())
	n.enque(2)
	n.enque(3)
	n.enque(4)
	n.enque(5)
	n.enque(6)
	n.display()
	fmt.Println(n.deque())
	n.display()
	n.top()
}
